utils/post_processing/bfiq_to_rawacf.py

```diff
@@ -0,0 +1,157 @@
+				intf_corrs[seq, beam] = np.outer(intf_samps.conjugate(), intf_samps)
+				cross_corrs[seq, beam] = np.outer(main_samps.conjugate(), intf_samps)
+
+				beam_offset = num_beams * num_ranges * num_lags
+				first_range_offset = int(ts_dict["first_range"] / ts_dict["range_sep"])
+
+				# Select out the lags for each range gate
+				main_small = np.zeros((num_ranges, num_lags,), dtype=np.complex64)
+				intf_small = np.zeros((num_ranges, num_lags,), dtype=np.complex64)
+				cross_small = np.zeros((num_ranges, num_lags,), dtype=np.complex64)
+
+				for rng in range(num_ranges):
+					for lag in range(num_lags):
+						# tau spacing in us, sample rate in hz
+						tau_in_samples = np.ceil(ts_dict["tau_spacing"] * 1e-6 * 
+													ts_dict["rx_sample_rate"])
+						p1_offset = lags[lag, 0] * tau_in_samples
+						p2_offset = lags[lag, 1] * tau_in_samples
+						
+						row_offset = int(rng + first_range_offset + p1_offset)
+						col_offset = int(rng + first_range_offset + p2_offset)
+
+						main_small[rng, lag] = main_corrs[seq, beam, row_offset, col_offset]
+						intf_small[rng, lag] = intf_corrs[seq, beam, row_offset, col_offset]
+						cross_small[rng, lag] = cross_corrs[seq, beam, row_offset, col_offset]
+
+				# replace full correlation matrix with resized range-lag matrix
+				out_main[seq, beam] = main_small
+				out_intf[seq, beam] = intf_small
+				out_cross[seq, beam] = cross_small
+
+		# average each correlation matrix over sequences dimension
+		out_main = np.mean(out_main, axis=0)
+		out_intf = np.mean(out_intf, axis=0)
+		out_cross = np.mean(out_cross, axis=0)
+
+		return out_main, out_intf, out_cross
+
+	warnings.simplefilter('ignore')
+
+	acf_file = bfiq_filepath.split('b')[-2] + 'rawacf.hdf5.test' 
+	temp_file= "temp_acf.hdf5"
+
+	bfiq = dd.io.load(bfiq_filepath)
+	acfs = dict()
+
+	shared_fields = ['beam_azms', 'beam_nums', 'blanked_samples', 'lags', 'noise_at_freq',
+	 					'pulses', 'sqn_timestamps', 'borealis_git_hash', 
+	 					'data_normalization_factor', 'experiment_comment', 
+	 					'experiment_id', 'experiment_name', 'first_range', 
+	 					'first_range_rtt', 'freq', 'int_time', 'intf_antenna_count', 
+	 					'main_antenna_count', 'num_sequences', 'num_slices', 'range_sep', 
+	 					'rx_sample_rate', 'samples_data_type', 'scan_start_marker', 
+	 					'slice_comment', 'station', 'tau_spacing', 'timestamp_of_write', 'tx_pulse_len']
+
+	for k in bfiq:
+		acfs[k] = dict()
+		# write common dictionary fields first
+		for f in shared_fields:
+			acfs[k][f] = bfiq[k][f]
+
+		# Perform correlations and write to dictionary
+		main_acfs, intf_acfs, xcfs = correlate_samples(bfiq[k])
+
+		acfs[k]["correlation_dimensions"] = np.array(main_acfs.shape)
+		acfs[k]["correlation_descriptors"] = np.array(['num_beams', 'num_ranges', 'num_lags'])
+
+		acfs[k]["main_acfs"] = main_acfs.flatten()
+		acfs[k]["intf_acfs"] = intf_acfs.flatten()
+		acfs[k]["xcfs"] = xcfs.flatten()
```

Comment:
technically, an xcf is not an acf, so if you were to be pedantic about this you could rename the 'acfs' to 'correlations' or something.